packages/relay/src/lib/services/metricService/metricService.ts

```diff
@@ -0,0 +1,355 @@
+   */
+  constructor(
+    logger: Logger,
+    sdkClient: SDKClient,
+    mirrorNodeClient: MirrorNodeClient,
+    hbarLimiter: HbarLimit,
+    register: Registry,
+    eventEmitter: EventEmitter,
+  ) {
+    this.logger = logger;
+    this.sdkClient = sdkClient;
+    this.hbarLimiter = hbarLimiter;
+    this.eventEmitter = eventEmitter;
+    this.mirrorNodeClient = mirrorNodeClient;
+    this.consensusNodeClientHistogramCost = this.initCostMetric(register);
+    this.consensusNodeClientHistogramGasFee = this.initGasMetric(register);
+
+    //listen to EXECUTE_TRANSACTION event to kick off captureTransactionMetrics() process
+    this.eventEmitter.on(constants.EVENTS.EXECUTE_TRANSACTION, (args: IExecuteTransactionEventPayload) => {
+      this.captureTransactionMetrics(args);
+    });
+
+    //listen to EXECUTE_QUERY event to kick off addExpenseAndCaptureMetrics() process
+    this.eventEmitter.on(constants.EVENTS.EXECUTE_QUERY, (args: IExecuteQueryEventPayload) => {
+      this.addExpenseAndCaptureMetrics(args);
+    });
+  }
+
+  /**
+   * Captures and logs transaction metrics by retrieving transaction records from the appropriate source
+   * and recording the transaction fees, gas usage, and other relevant metrics.
+   *
+   * @param {IExecuteTransactionEventPayload} payload - The payload object containing transaction details.
+   * @param {string} payload.requestId - The unique identifier for the request.
+   * @param {string} payload.callerName - The name of the entity calling the transaction.
+   * @param {string} payload.transactionId - The unique identifier for the transaction.
+   * @param {string} payload.transactionType - The type of the transaction being processed.
+   * @param {string} payload.txConstructorName - The name of the transaction constructor.
+   * @param {string} payload.operatorAccountId - The account ID of the operator managing the transaction.
+   * @param {string} payload.interactingEntity - The entity interacting with the transaction.
+   * @returns {Promise<void>} - A promise that resolves when the transaction metrics have been captured.
+   */
+  public async captureTransactionMetrics({
+    requestId,
+    callerName,
+    transactionId,
+    transactionType,
+    txConstructorName,
+    operatorAccountId,
+    interactingEntity,
+  }: IExecuteTransactionEventPayload): Promise<void> {
+    // retrieve metrics
+    const transactionRecordMetrics = await this.getTransactionRecordMetrics(
+      transactionId,
+      callerName,
+      requestId,
+      txConstructorName,
+      operatorAccountId,
+    );
+
+    // capture metrics to HBAR rate limiter and metric registry
+    if (transactionRecordMetrics) {
+      const { gasUsed, transactionFee, txRecordChargeAmount } = transactionRecordMetrics;
+      if (transactionFee !== 0) {
+        this.addExpenseAndCaptureMetrics({
+          executionType: `TransactionExecution`,
```

Comment:
Please, define an enum with possible ExecutionTypes inside the `MetricService` and use it here:
```suggestion
          executionType: ExecutionType.Transaction,
```

Reply:
I like this idea! Updated.